src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JunctionTreeKBestHaplotypeFinder.java

```diff
@@ -0,0 +1,198 @@
+
+            } else {
+                // We have already expanded this part of the graph, use it.
+                vertexToExtend = graph.getEdgeTarget(chain.get(chain.size() - 1));
+                outgoingEdges = graph.outgoingEdgesOf(vertexToExtend);
+            }
+            // vertexToExtend and outgoingEdges are necessarily at the next "interesting" point
+
+
+            ////////////////////////////////////////////////////////////
+            // code to decide what to do at that interesting node
+            ////////////////////////////////////////////////////////////
+            // In the event we have a junction tree on top of a vertex with outDegree > 1, we add this first before we traverse paths
+            if ( experimentalReadThreadingGraph.getJunctionTreeForNode((MultiDeBruijnVertex) vertexToExtend) != null) { //TODO make the condition for this actually based on the relevant junction tree
+                // TODO chain can be null but we still need to inherit a thing, probably happens whenever we pick up a tree.
+                // ignore starting junction tree
+                pathToExtend.addJunctionTree(experimentalReadThreadingGraph.getJunctionTreeForNode((MultiDeBruijnVertex) vertexToExtend));
+            }
+
+            //TODO this can probabaly be 100% consumed by getApplicableNextEdgesBasedOnJunctionTrees() as a check... that would simplify things somewhat
+            // If we are at a reference end then we close out the path
+            if (sinks.contains(vertexToExtend) && pathToExtend.hasStoppingEvidence(weightThresholdToUse)) {
+                //TODO this will probably be resolved using a junction tree on that node and treating it as an edge to extend
+                //todo the proposal here would be to check if there is an active tree left for us at this point and if so keep going
+                if (chain.isEmpty()) {
+                    result.add(pathToExtend);
+                } else {
+                    result.add(new JTBestHaplotype<>(pathToExtend, chain, 0));
+                }
+            }
+            // NOTE: even if we are at the reference stop and there is evidence in the junction trees of a stop we still want to explore other edges potentially
+
+            // We must be at a point where the path diverges, use junction trees to resolve if possible
+            if (outgoingEdges.size() > 1) {
+                List<JTBestHaplotype<V, E>> jTPaths = pathToExtend.getApplicableNextEdgesBasedOnJunctionTrees(chain, outgoingEdges, weightThresholdToUse);
+                if (jTPaths.isEmpty() && !sinks.contains(vertexToExtend)) {
+//                    throw new GATKException("Found no path based on the junction trees or exisiting paths, this should not have happened");
+                    System.out.println("Found nothing Queue has this many: "+queue.size()+"\nPath that failed to extend was junction tree: "+pathToExtend.getVertices());
```

Comment:
hmm... This is a debug statement for an older iteration of the code. Its interesting looking back at my logs it appears these are happening more often than I expected, given than these are sites with multiple outgoing paths and yet it fails to follow any path... I'm going to add this to my todo backlog because I think there is some failure state involving loops. @jamesemery

Reply:
Its theoretically possible it comes up from looping reference paths that don't get covered by reads sufficiently but i'm not sure I believe that would be common enough to explain this.